### PR TITLE
Suppress lib update warning

### LINKED ISSUE #1825: Allow users to disable dependency updates
**Is your feature request related to a problem? Please describe.**
Currently if a user wants to keep a certain version of the dependency they will get a warning that it's outdated, but it might not always be possible to update it.

**Describe the solution you'd like**
Let's at a very minimum add and option to disable these warnings, we can follow up with something more concrete

**Describe alternatives you've considered**
Disable all the actionable diagnostics with `scala-cli config actions false`

**Additional context**
https://github.com/VirtusLab/scala-cli/issues/1822

### PR DESCRIPTION
**Pull request by MaciejG604**
Resolves #1825

### CODE REVIEW

File: modules/cli/src/main/scala/scala/cli/commands/shared/SharedOptions.scala [hunk 1]
```diff
@@ -432,8 +442,8 @@ final case class SharedOptions(
     }
   }
 
-  def suppressDirectivesInMultipleFilesWarning: Option[Boolean] =
-    suppress.suppressDirectivesInMultipleFilesWarning.filter(identity).orElse(
+  def getOptionOrFromConfig(cliOption: Option[Boolean], configDbKey: BooleanEntry) =
+    cliOption.orElse(
       configDb.map(_.get(Keys.suppressDirectivesInMultipleFilesWarning))
```
Comment (lwronski) [inline modules/cli/src/main/scala/scala/cli/commands/shared/SharedOptions.scala]:
```suggestion
      configDb.map(_.get(configDbKey))
```

File: modules/options/src/main/scala/scala/build/actionable/ActionableDependencyHandler.scala [hunk 2]
```diff
@@ -25,30 +25,35 @@ case object ActionableDependencyHandler
...
-      if (dependency.userParams.contains("toolkit"))
-        Some(ActionableDependencyUpdateDiagnostic(
-          setting.positions,
-          currentVersion,
-          latestVersion,
-          dependencyModuleName = "toolkit",
-          suggestion = latestVersion
-        ))
+  ): Either[BuildException, Option[ActionableDependencyUpdateDiagnostic]] =
+    if (buildOptions.suppressWarningOptions.suppressOutdatedDependencyWarning.getOrElse(false))
```
Comment (lwronski) [inline modules/options/src/main/scala/scala/build/actionable/ActionableDependencyHandler.scala]: Can you move this condition higher? to the `extractSettings` function.

**Review by lwronski — APPROVED** [review]

**Review by Gedochao — APPROVED** [review]

**Comment by SethTisue** [comment]
Is there a way to suppress these warnings by adding something to the `.scala` file itself, either per-dependency or project-wide? I'd rather not suppress them globally, but I also don't want to have to enter the flag on the command line every time.

**Comment by tgodzik** [comment]
> Is there a way to suppress these warnings by adding something to the `.scala` file itself, either per-dependency or project-wide? I'd rather not suppress them globally, but I also don't want to have to enter the flag on the command line every time.

Unfortunately, those are the only two options currently. It might be hard to figure out an API to do it per case basis.

**Pull request merged**